libr/debug/p/native/linux/linux_debug.c

```diff
@@ -386,8 +391,8 @@ RDebugReasonType linux_dbg_wait(RDebug *dbg, int my_pid) {
 	// Ignore keyboard interrupt while waiting to avoid signaling the child process twice on
 	// break(SIGSTOP is sent by wait_break) which forced the user to continue an extra time
 	// to handle SIGINT
-	r_sys_signal (SIGINT, SIG_IGN);
-	r_cons_break_push ((RConsBreak)linux_dbg_wait_break, dbg);
+	// r_sys_signal (SIGINT, SIG_IGN);
```

Comment:
Anyway, I believe using `r_sys_signal` here is still wrong, especially at the end it calls `r_sys_signal (SIGINT, SIG_DFL)` which restore the default action when receiving `SIGINT` back to terminating the process. Therefore, interrupting `r2` with ^C next time will exit it instead of calling the callback in `r_cons_breaked_stack`. `r_cons` already set the default action for `SIGINT` to be `__break_signal` in `r_cons.c`, so other code should not change that behavior.

In addition, `linux_dbg_wait_break` callback does not always work in situations when the debugee is in a stopped state. For example, interrupting `r2` will call `linux_dbg_wait_break` which sends the `SIGSTOP` to the debugee, but in our case we are calling `r_debug_step` repeatedly so the signal will not be delivered to the process. Then, the next time we continue the debugee, it will stop immediately because of the pending signal.

Reply:
This will be fixed in another PR.